pandas/core/window/rolling.py

```diff
@@ -421,26 +420,40 @@ def _apply_blockwise(
             # GH 12541: Special case for count where we support date-like types
             obj = notna(obj).astype(int)
             obj._mgr = obj._mgr.consolidate()
-        mgr = obj._mgr
 
-        def hfunc(bvalues: ArrayLike) -> ArrayLike:
-            # TODO(EA2D): getattr unnecessary with 2D EAs
-            values = self._prep_values(getattr(bvalues, "T", bvalues))
-            res_values = homogeneous_func(values)
-            return getattr(res_values, "T", res_values)
-
-        def hfunc2d(values: ArrayLike) -> ArrayLike:
+        def hfunc(values: ArrayLike) -> ArrayLike:
             values = self._prep_values(values)
             return homogeneous_func(values)
 
-        if isinstance(mgr, ArrayManager) and self.axis == 1:
-            new_mgr = mgr.apply_2d(hfunc2d, ignore_failures=True)
-        else:
-            new_mgr = mgr.apply(hfunc, ignore_failures=True)
+        if self.axis == 1:
+            obj = obj.T
 
-        if 0 != len(new_mgr.items) != len(mgr.items):
+        taker = []
+        res_values = []
+        for i, arr in enumerate(obj._iter_column_arrays()):
+            # GH#42736 operate column-wise instead of block-wise
+            try:
+                res = hfunc(arr)
+            except (TypeError, NotImplementedError):
```

Comment:
Will we allow these to raise once the nuisance column deprecation is enforced?

Reply:
yes